src/reducers/graphReducer.ts

```diff
@@ -35,6 +35,16 @@ const slice = createSlice({
       state.nodes = [...state.nodes, ...newNodes];
       return state;
     },
+    updateNode: (state, action) => {
+      const { updateNodeId, updated } = action.payload;
+      const stateNodeIndex = state.nodes.findIndex(node => node.id === updateNodeId);
+      if (stateNodeIndex !== -1) {
+        state.nodes[stateNodeIndex] = { ...state.nodes[stateNodeIndex], ...updated};
```

Comment:
We want to clone the state first, before modifying it. Redux likes to keep state immutable. Then we return the modified state at the end of the fuction.

Reply:
Turns out that the redux-toolkit being use incorporates immer, which makes keeping the state immutable unnecessary. We can modify the state objects and immer with handle cloning the state for us. This is fine as is.